### PR TITLE
feat(radio checkbox): add tokens/update shim

### PR DESCRIPTION
**Pull request by valentin-mladenov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/angular/src/forms/checkbox/STYLES.md [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
+| --clr-forms-checkbox-background-color           | Background color of checkbox                    |
+| --clr-forms-checkbox-indeterminate-border-color | Border color of indeterminate checkbox          |
+| --clr-forms-checkbox-mark-color                 | Color of checkbox mark                          |
+| --clr-forms-checkbox-disabled-background-color  | Background color of disabled checkbox           |
+| --clr-forms-checkbox-disabled-mark-color        | Color of disabled checkbox mark                 |
+| --clr-forms-checkbox-border-radius              | Border radius of checkbox                       |
+| --clr-forms-checkbox-line-height                | The line height of the checkbox button wrapper. |
+| --clr-forms-checkbox-input-top                  | The top space for the checkbox input element.   |
+| --clr-forms-checkbox-padding-left               | The left padding for checkbox label element.    |
+| --clr-forms-checkbox-label-height               | The height of the checkbox label element.       |
```
Comment (dtsanevmw) [inline projects/angular/src/forms/checkbox/STYLES.md:17]:
```suggestion
| --clr-forms-checkbox-line-height                | Line height of the checkbox button wrapper |
| --clr-forms-checkbox-input-top                  | Top position for the checkbox input element   |
| --clr-forms-checkbox-padding-left               | Left padding for checkbox label element    |
| --clr-forms-checkbox-label-height               | Height of the checkbox label element       |
```

File: projects/angular/src/forms/radio/STYLES.md [hunk 2]
```diff
@@ -7,6 +7,10 @@
 | --clr-forms-radio-label-color               | The color of the label for radio buttons.                     |
 | --clr-forms-radio-disabled-background-color | The background color of disabled radio buttons.               |
 | --clr-forms-radio-disabled-mark-color       | The color of the mark (indicator) for disabled radio buttons. |
+| --clr-forms-radio-line-height               | The line height of the radio button wrapper.                  |
+| --clr-forms-radio-input-top                 | The top space for the radio input element.                    |
+| --clr-forms-radio-padding-left              | The left padding for radio label element.                     |
+| --clr-forms-radio-label-height              | The height of the radio label element.                        |
```
Comment (dtsanevmw) [inline projects/angular/src/forms/radio/STYLES.md:13]:
```suggestion
| --clr-forms-radio-line-height               | Line height of the radio button wrapper                  |
| --clr-forms-radio-input-top                 | Top space for the radio input element                    |
| --clr-forms-radio-padding-left              | Left padding for radio label element                     |
| --clr-forms-radio-label-height              | Height of the radio label element                        |
```

File: projects/angular/src/forms/styles/_mixins.forms.scss [hunk 3]
```diff
@@ -118,12 +118,16 @@
   cursor: not-allowed;
 }
 
-@mixin checkbox-radio-input-styles($size) {
+@mixin checkbox-radio-input-styles(
+  $size,
+  $input-top: clr-forms-radio-input-top,
+  $input-top-fallback: forms-variables.$clr-forms-radio-input-top
```
Comment (dtsanevmw) [inline projects/angular/src/forms/styles/_mixins.forms.scss:124]: What's the use of this variable if we use it only for the hidden input? If we are going to allow different top positioning of the checkbox then it should affect both the hidden and the visible checkbox.

File: projects/angular/src/forms/styles/_checkbox.clarity.scss [hunk 4]
```diff
@@ -106,8 +112,8 @@
       border-left: none;
       @include mixins.css-var(
         border-bottom-color,
-        $indetermCssBrdrColorVar,
-        $indetermSassBrdrColorVar,
+        clr-forms-checkbox-indeterminate-border-color,
+        forms-variables.$clr-forms-checkbox-indeterminate-border-color,
```
Comment (dtsanevmw) [inline projects/angular/src/forms/styles/_checkbox.clarity.scss:116]: Looks like both of these are `indeterminate` so they should use the same variable

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
I added some suggestions and format fixes for styles documentation.

File: projects/angular/src/forms/styles/_checkbox.clarity.scss [hunk 5]
```diff
@@ -88,15 +96,13 @@
...
     input[type='checkbox']:indeterminate + .clr-control-label::before {
       border: variables.$clr-global-borderwidth solid;
       @include mixins.css-var(
         border-color,
-        $indetermCssBrdrColorVar,
-        $indetermSassBrdrColorVar,
+        clr-forms-checkbox-border-color,
+        forms-variables.$clr-forms-checkbox-indeterminate-border-color,
         variables.$clr-use-custom-properties
       );
```
Comment (dtsanevmw) [inline projects/angular/src/forms/styles/_checkbox.clarity.scss:107]: Looks like both of these are `indeterminate` so they should use the same variable. `clr-forms-checkbox-border-color` is not needed here.

Comment (valentin-mladenov) [reply projects/angular/src/forms/styles/_checkbox.clarity.scss]: corrected in commit b3df0d8

File: projects/angular/src/forms/radio/STYLES.md (hunk 2 above)
Comment (valentin-mladenov) [reply projects/angular/src/forms/radio/STYLES.md]: corrected in commit e0847ba

File: projects/angular/src/forms/checkbox/STYLES.md (hunk 1 above)
Comment (valentin-mladenov) [reply projects/angular/src/forms/checkbox/STYLES.md]: corrected in commit e0847ba

File: projects/angular/src/forms/styles/_mixins.forms.scss (hunk 3 above)
Comment (valentin-mladenov) [reply projects/angular/src/forms/styles/_mixins.forms.scss:124]: The input is opaque.
In it's initial state was hard coded to `top: variables.$clr_baselineRem_5px;` I decided to parameterize so the hidden can overlap with the visible checkbox with tokens.
The visible and the hidden checkbox cannot use same parameter unless we introduce some very complicated calculation for it.
I don't think there is much to gain from the top parameter.
@dtsanevmw I can revert my change.

Comment (valentin-mladenov) [reply projects/angular/src/forms/styles/_mixins.forms.scss]: corrected in commit a7e1db6

File: projects/angular/src/forms/styles/_checkbox.clarity.scss [hunk 6]
```diff
@@ -106,14 +108,27 @@
...
       display: inline-block;
       transform: translate(0, $mark-half-size);
     }
 
+    input[type='checkbox'].clr-indeterminate:checked + .clr-control-label::after,
+    input[type='checkbox']:indeterminate:checked + .clr-control-label::after {
+      border-left: forms-variables.$clr-forms-checkbox-mark-border-size solid;
+      border-bottom: forms-variables.$clr-forms-checkbox-mark-border-size solid;
```
[the remark this answers is not in the document]
Comment (valentin-mladenov) [reply projects/angular/src/forms/styles/_checkbox.clarity.scss:122]: There are no tokens that they can be map to. `forms-variables.$clr-forms-checkbox-mark-border-size` is related to the inner check mark ☑️ of the checkbox. The design doesn't specify anything about it.

[the remark this answers is not in the document]
Comment (kevinbuhmann) [reply projects/angular/src/forms/styles/_checkbox.clarity.scss:122]: Okay, cool. Just wanted to double check.

**Comment by valentin-mladenov** [comment]
Depends on #812

File: projects/ui/src/shim.cds-core.scss [hunk 7]
```diff
@@ -472,17 +472,34 @@
...
+  --clr-forms-checkbox-line-height: var(--cds-global-space-9);
+  --clr-forms-checkbox-padding-left: var(--cds-global-space-9);
+  --clr-forms-checkbox-label-height: var(--cds-global-space-9);
 
   // Radio
-  --clr-forms-radio-label-color: var(--clr-forms-checkbox-label-color); // indirectly mapped to CDS token
-  --clr-forms-radio-disabled-background-color: var(--clr-form-disabled-background-color); // indirectly mapped to CDS token
-  --clr-forms-radio-disabled-mark-color: var(--clr-forms-checkbox-disabled-mark-color); // indirectly mapped to CDS token
+  --clr-forms-radio-label-color: var(--clr-forms-label-color); // indirectly mapped to CDS token --cds-global-typography-color-400
+  --clr-forms-radio-label-disabled-color: var(--cds-alias-typography-disabled, var(--cds-global-typography-disabled));
```
Comment (dtsanevmw) [inline projects/ui/src/shim.cds-core.scss:493]: Should be available trough the added from Will file `token_shim.cds-core.scss` so it no longer needs that fallback.

File: projects/angular/src/forms/styles/_checkbox.clarity.scss [hunk 8]
```diff
@@ -106,14 +108,27 @@
...
+      @include mixins.css-var(
+        border-color,
+        clr-forms-checkbox-mark-color,
+        forms-variables.$clr-forms-checkbox-mark-color,
+        variables.$clr-use-custom-properties
+      );
+      transform: translate(0, $mark-half-size) rotate(-45deg);
+    }
```
Comment (dtsanevmw) [inline projects/angular/src/forms/styles/_checkbox.clarity.scss:130]: Is that needed? Basically it doesn't matter for the checked state if it's `indeterminate` or not and it's already working without these lines.

**Review by dtsanevmw — CHANGES_REQUESTED** [review]
One small token change proposal and as discussed offline, remove the css with having `:checked:indeterminate` as we can't have both states at once.

File: projects/angular/src/forms/styles/_checkbox.clarity.scss (hunk 8 above)
Comment (valentin-mladenov) [reply projects/angular/src/forms/styles/_checkbox.clarity.scss]: corrected in commit: 628b080

File: projects/ui/src/shim.cds-core.scss (hunk 7 above)
Comment (valentin-mladenov) [reply projects/ui/src/shim.cds-core.scss]: corrected in commit: 628b080

**Review by dtsanevmw — APPROVED** [review]
LGTM!

**Pull request merged**